tests/virt_autotest/guest_migration_src.pm

```diff
@@ -83,6 +83,37 @@ sub analyzeResult {
     return $result;
 }
 
+#Mutex lock/unlock functionality
+sub set_mutex_lock {
+    my ($self, $lock_signal) = @_;
+
+    mutex_lock($lock_signal);
+    mutex_unlock($lock_signal);
+}
+
+sub post_execute_script_configuration {
+    my $self = shift;
+
+    #let dst host upload logs
+    set_var('SRC_TEST_DONE', 1);
+    bmwqemu::save_vars();
+    $self->set_mutex_lock('DST_UPLOAD_LOG_DONE');
+}
+
+sub post_execute_script_assertion {
+    my $self = shift;
+
+    #display test result
+    $self->{script_output} = script_output("cd /tmp; zcat $self->{compressed_log_name}.tar.gz | sed -n '/Executing check validation/,/[0-9]* fail [0-9]* succeed/p'");
+    save_screenshot;
+
+    my $output = $self->{script_output};
+
+    my $guest_migration_src_assert_pattern = "[1-9]{1,}[[:space:]]fail|[1-9]{1,}[[:space:]]internal_error";
```

Comment:
@waynechen55 How do you figure out this src assert pattern? I checked the original src file, it does not use it for assert. The original logic for assert is that if such assertion fails, test can die and go to post_fail_hook of its parent class recursively until finding an unempty one.

@Julie-CAO You are the owner of prj3. Please help to double confirm.

Reply:
This is almost the same to the original pattern, previously it matches "0 succeed" now it matches fails and error directly. Now it is more straight forward. Previous it just dies immediately without calling any post_fail_hook

Reply:
From logic view, I am afraid the reverse of "0 succeed" is not easy to be covered fully, at least it does not equal to "[1-9]{1,}[[:space:]]fail|[1-9]{1,}[[:space:]]internal_error". Better to think twice here. BTW it is ctcs2 output, I am not sure how reliable the fail/internal_error count is, at least I never watch them for various test status and check them. Be careful here!

Reply:
1. I am almost sure "[1-9]{1,}[[:space:]]fail|[1-9]{1,}[[:space:]]internal_error" covers all we need to cover, because according to my observation for a long time,  if it is a fully pass, there will never be any fail or internal error. I have never seen ctcs2 output misinterpret any test result. By the way, if you say logic, for exmaple, if $output =~ "0 succeed", then die is definitely not very complete, because it will cover partially failure, for example, 1 succeed and 1 failure.
2. OK, now it just uses its direct base class one.